### ISSUE TITLE
Add ROR (Research Organization Registry) as Funding Body vocab source

### ISSUE THREAD

**Issue by carakey** [osulp/Scholars-Archive#2604]
### Descriptive summary

We formerly had tickets for adding Crossref / Fundref / Funder Registry support for Funding Body (#586, #587). ROR is [replacing Fundref](https://ror.org/blog/2023-09-07-open-funder-registry-transition-ror-cross-post/). We should build out the Funding Body field using ROR as a vocab source, especially if it can work with Questioning Authority. 

This would help support Open Scholarship initiatives.

See https://ror.org/registry/ for details.

### Expected behavior

Authorized headings for funding agencies can be pulled from ROR into the Funding Body field, ideally with type-ahead functionality. 

### Related work

#586 
#587

**Comment by lamtu1** [comment]
Hi @carakey, I was wondering about this `ror` thing, do you want the typeahead to be able to search by `word`, `uri`, or `id`? Or maybe even a combination of both or more?

**Comment by carakey** [comment]
@lamtu1 I think that `word` is the most user-friendly for the majority of depositors, so that would be my top priority. If we can have more than one typeahead or entry option, then I could see some users preferring `id`. Thanks.

**Comment by lamtu1** [comment]
Hi @carakey, I forgot to ask about this since the `funding_body` is moving to `ROR` and the current predicate for it is this: `::RDF::Vocab::MARCRelators.fnd`. I was wondering does the `funding_body` needs an update to the predicate with something new from `ROR` or we kept the predicate as it is and implement `ROR` as a search functionality? Thank you

**Comment by carakey** [comment]
@lamtu1 No this ticket isn't for replacing the predicate, but it's for more than the search functionality. __This is to set ROR as the source vocabulary for the field value.__ This is the same as what was laid out in #586, except that was for a different, obsolete vocabulary. 

Specifically, similar to other controlled, URI-typed fields. Ideally:`funding_body` should store a URI like `https://ror.org/035a68863`, display a label on the work page like `United States Geological Survey`, and do a type-ahead in the form to make it straightforward for depositors.

**Comment by carakey** [comment]
**QA:**

_QA was performed editing two existing works in different work types, [One](https://ir-staging.library.oregonstate.edu/concern/technical_reports/3x816m825) and [Two](https://ir-staging.library.oregonstate.edu/concern/undergraduate_thesis_or_projects/9019s273j) on staging. Results were the same for both._

**TL;DR** The functionality works within the metadata form, but the ROR term is not getting saved to the work.

* Criteria: The Funding Body field in the metadata form enables searching for and selecting a ROR term
    * Pass! I searched "NSF" and several results were provided, including the one I intended (U.S. National Science Foundation)
* Bonus Criteria: The Funding Body field in the metadata accepts a known ROR URI
    * Unsure? I entered OSU's ROR `https://ror.org/00ysfqy60` as a second Funding Body, on the second QA work only. It did not give an error on save. Since the terms are not being stored, I'm not sure if this works or not. 
* Criteria: The ROR URI is stored for the work (checking NT and/or SOLR for this)
    * **FAIL** -- I downloaded the `.nt` and there was no Funding Body field or ROR URI present. 
* Criteria: The ROR label displays on the work show page 
    * **FAIL** -- No Funding Body field is displayed (not unexpected given it wasn't saved to the metadata)

**Comment by lamtu1** [comment]
`QA`: This have to QA again from the beginning since the functionality got updated

**Comment by carakey** [comment]
**QA:**

_QA was performed editing two existing works in different work types, a [Tech Report](https://ir-staging.library.oregonstate.edu/concern/technical_reports/9k41zd81j/) and [Dataset](https://ir-staging.library.oregonstate.edu/concern/datasets/tt44pn099) on staging. Results were the same for both._


* Criteria: The Funding Body field in the metadata form enables searching for and selecting a ROR term
    * Pass! I searched for a string in the name as well as an acronym and several results were provided in each case. 
* Criteria: The ROR URI is stored for the work (checking NT and/or SOLR for this)
    * Pass! `<https://ir-staging.library.oregonstate.edu/concern/datasets/tt44pn099> <http://id.loc.gov/vocabulary/relators/fnd> <https://ror.org/021nxhr62> .` (for QA item no. two) 
* Criteria: The ROR label displays on the work show page 
    * Pass! The label appears on the page correctly and the "more info" link goes to the right destination.

QA Pass!
